crates/span/src/map.rs

```diff
@@ -13,6 +13,7 @@ use crate::{
 /// Maps absolute text ranges for the corresponding file to the relevant span data.
 #[derive(Debug, PartialEq, Eq, Clone, Hash)]
 pub struct SpanMap<S> {
+    /// The offset stored here is the *end* of the node.
```

Comment:
I believe my thinking was that if using the start we'd need to separately store the end of the final node/length of the file while also always having a `0` at the start. So encoding the end is slightly more optimal. Either way this should've been documented, sorry.